app/main/validators.py

```diff
@@ -226,3 +226,27 @@ def __call__(self, form, field):
                         raise StopValidation("Your file contains a virus")
                 finally:
                     field.data.seek(0)
+
+
+class Only2DecimalPlaces:
+    regex = re.compile(r"^[0-9]+(?:\.[0-9]{1,2})?$")
+
+    def __init__(self, message="Enter a value with 2 decimal places."):
+        self.message = message
+
+    def __call__(self, form, field):
+        if field.data and not re.match(self.regex, str(field.data)):
+            raise ValidationError(self.message)
+
+
+class IsPostcode:
+    regex = re.compile(
+        r"^([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z])))) [0-9][A-Za-z]{2})$"  # noqa: E501
```

Comment:
This probably fails if they don't include the space between `SW1A 1AA`, you can also simplify it greatly by making it case insensitive (which turns `[A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2}` into `[a-z][a-hj-y][0-9]{1,2}]`

Personally though I would just use a library https://pypi.org/project/postcodes-uk/, tht way you can leave the unit testing to them and not worry about maintaining it

Reply:
Using postcode-validator now instead for regex for validation & for formatting postcodes.